code/modules/awaymissions/zlevel.dm

```diff
@@ -18,39 +54,80 @@
 
 		var/pos = findtext(t, " ")
 		var/name = null
-	//	var/value = null
 
 		if (pos)
 			name = lowertext(copytext(t, 1, pos))
-		//	value = copytext(t, pos + 1)
+
 		else
 			name = lowertext(t)
 
 		if (!name)
 			continue
 
-		potentialRandomZlevels.Add(t)
+		potentialMaps.Add(t)
 
+		return potentialMaps
 
-	if(potentialRandomZlevels.len)
-		world << "<span class='boldannounce'>Loading away mission...</span>"
 
-		var/map = pick(potentialRandomZlevels)
+/proc/seedRuins(z_level = 1, ruin_number = 0, whitelist = /area/space, list/potentialRuins = potentialSpaceRuins)
+	if(ruin_number > potentialRuins.len)
+		ruin_number = potentialRuins.len //To avoid someone doing something dumb and entering an infinite loop

```

Comment:
nitpick: i think this would be more readable as `ruin_number = min(ruin_number, potentialRuins.len)`